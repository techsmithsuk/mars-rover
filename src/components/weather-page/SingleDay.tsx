import React from 'react';
import {SingleDayData} from './SingleDayData';
import './singleDay.scss';

import ThermometerIcon from './resources/svg_icon_thermometer';
import CloudIcon from './resources/svg_icon_cloud';
import SunIcon from './resources/svg_icon_sun';
import WindIcon from './resources/svg_icon_wind';

export function SingleDaySummary(props : SingleDayData) {
    
    let avTemp = props.averageTemp ? Math.round(props.averageTemp) : 0;   

    return (
        <div className="SingleDaySummary">            

            <div className="selectionTab"></div>

            <div className="weekday">
                <p>{props.weekdayName}</p>
            </div>

            <div className="icon">
                <SunIcon/> 
            </div>
            
            <div className="AverageTemp">
                {avTemp}                   
            </div>   
                      
        </div>
    )
}

export function SingleDayDetail(props : SingleDayData) {  
    
    let min = props.minTemp ? Math.round(props.minTemp) : 0;   
    let max = props.maxTemp ? Math.round(props.maxTemp) : 0; 

    let date = props.earthSolDate;   

    return (
        <div className="SingleDayDetail">            

            <div className="weekdayBlock">
                <p>DATE</p>
                <p>{props.weekdayName} {props.earthSolDate}</p>
                <p>MARS SOL {props.marsSolDate}</p>
                

            </div> 

            <div className="tempBlock">
                <ThermometerIcon/>  
                <p>High: {max}</p>
                <p>Low: {min}</p>
                
            </div>

            <div className="windBlock">
                <WindIcon/>  
                <p>{props.averageWindSpeed} mph</p>
                <p>{props.windDirection}</p>    
            </div>

            <p className="textBlock">
                "Sunny with a chance of meatballs."
            </p>         
                      
        </div>
    )
}