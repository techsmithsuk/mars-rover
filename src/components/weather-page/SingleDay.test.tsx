import React from 'react';
import renderer from 'react-test-renderer';
import { SingleDayDetail, SingleDaySummary } from './SingleDay';
import { SingleDayData } from './SingleDayData';

describe("Single Day Summary Component", () => {
    it("Should render without errors", () => {
        const testData: SingleDayData = {  
            marsSolDate: '344',
            earthSolDate: '14th',
            averageTemp: -67.108,
            minTemp: -99.926,
            maxTemp: -23.287,
            averageWindSpeed: 5.3,
            windDirection: 'SSE',
            averagePressure: 681.448,
            weekdayName: 'Monday' // TODO calc via function
        }

        const component = renderer.create(
            <SingleDaySummary {...testData}/>
        );
        
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Single Day Detail Component", () => {
    it("Should render without errors", () => {
        const testData: SingleDayData = {  
            marsSolDate: '344',
            earthSolDate: '14th',
            averageTemp: -67.108,
            minTemp: -99.926,
            maxTemp: -23.287,
            averageWindSpeed: 5.3,
            windDirection: 'SSE',
            averagePressure: 681.448,
            weekdayName: 'Monday' // TODO calc via function
        }

        const component = renderer.create(
            <SingleDayDetail {...testData}/>
        );
        
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});