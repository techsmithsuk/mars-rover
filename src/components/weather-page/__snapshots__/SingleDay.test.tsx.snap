// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Day Detail Component Should render without errors 1`] = `
<div
  className="SingleDayDetail"
>
  <div
    className="weekdayBlock"
  >
    <p>
      DATE
    </p>
    <p>
      Monday
       
      14th
    </p>
    <p>
      MARS SOL 
      344
    </p>
  </div>
  <div
    className="tempBlock"
  >
    <svg
      height="581pt"
      viewBox="-184 0 581 581.55153"
      width="581pt"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m107.15625 0c-37.46875.0429688-67.832031 30.40625-67.875 67.875v325.101562c-28.816406 23.519532-43.148438 60.460938-37.734375 97.261719 8.230469 58.328125 62.191406 98.9375 120.515625 90.703125 52.707031-7.441406 91.859375-52.601562 91.75-105.828125.097656-31.824219-14.148438-61.992187-38.785156-82.136719v-325.101562c-.039063-37.46875-30.402344-67.8320312-67.871094-67.875zm87.265625 475.113281c.0625 48.191407-38.953125 87.316407-87.148437 87.378907-48.195313.066406-87.320313-38.953126-87.382813-87.148438-.035156-27.507812 12.898437-53.421875 34.90625-69.925781 2.441406-1.832031 3.878906-4.707031 3.878906-7.757813v-29.207031h19.390625v-19.390625h-19.390625v-29.089844h19.390625v-19.390625h-19.390625v-29.089843h19.390625v-19.390626h-19.390625v-29.089843h19.390625v-19.390625h-19.390625v-29.089844h19.390625v-19.390625h-19.390625v-29.089844h19.390625v-19.390625h-19.390625v-38.785156c0-26.777344 21.703125-48.480469 48.480469-48.480469 26.773438 0 48.480469 21.703125 48.480469 48.480469v329.804688c0 3.050781 1.4375 5.925781 3.878906 7.753906 22.019531 16.382812 34.96875 42.234375 34.90625 69.679687zm0 0"
      />
      <path
        d="m116.851562 427.601562v-340.335937h-19.390624v340.335937c-26.234376 5.355469-43.160157 30.960938-37.804688 57.195313s30.960938 43.160156 57.195312 37.804687c26.234376-5.355468 43.160157-30.960937 37.804688-57.195312-3.886719-19.042969-18.765625-33.917969-37.804688-37.804688zm-9.695312 76.597657c-16.066406 0-29.089844-13.023438-29.089844-29.085938 0-16.066406 13.023438-29.089843 29.089844-29.089843s29.089844 13.023437 29.089844 29.089843c0 16.0625-13.023438 29.085938-29.089844 29.085938zm0 0"
      />
    </svg>
    <p>
      High: 
      -23
    </p>
    <p>
      Low: 
      -100
    </p>
  </div>
  <div
    className="windBlock"
  >
    <svg
      height="581pt"
      viewBox="0 0 512 512"
      width="581pt"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.979,271.181c30.288,25.979,64.616,55.426,131.017,55.426c66.403-0.001,100.733-29.447,131.022-55.426 c28.644-24.568,53.382-45.787,104.98-45.787c51.599,0,76.337,21.219,104.98,45.787c8.385,7.191,21.011,6.225,28.202-2.159 c7.191-8.385,6.225-21.011-2.16-28.202c-30.289-25.979-64.619-55.426-131.022-55.426S273.265,214.84,242.976,240.82 c-28.644,24.568-53.382,45.787-104.98,45.787c-51.597,0-76.332-21.218-104.974-45.787c-8.384-7.191-21.01-6.226-28.202,2.159 C-2.372,251.363-1.405,263.99,6.979,271.181z"
      />
      <path
        d="M6.979,161.787c30.287,25.98,64.615,55.426,131.017,55.426c66.403,0,100.733-29.446,131.022-55.426 C297.662,137.219,322.399,116,373.998,116c51.597,0,76.335,21.218,104.98,45.787c8.385,7.193,21.01,6.224,28.201-2.161 c7.192-8.384,6.223-21.012-2.161-28.202C474.729,105.444,440.399,76,373.998,76c-66.403,0-100.733,29.446-131.022,55.426 c-28.644,24.568-53.382,45.787-104.98,45.787c-51.596,0-76.332-21.218-104.974-45.787c-8.384-7.192-21.011-6.226-28.202,2.159 C-2.372,141.969-1.405,154.596,6.979,161.787z"
      />
      <path
        d="M505.02,350.213c-30.289-25.979-64.619-55.426-131.022-55.426s-100.733,29.446-131.022,55.426 C214.332,374.781,189.594,396,137.996,396c-51.594,0-76.331-21.218-104.973-45.785l-0.002-0.002 c-8.384-7.189-21.009-6.223-28.201,2.16c-7.191,8.384-6.224,21.011,2.16,28.202C37.268,406.555,71.597,436,137.996,436 c66.403,0,100.733-29.446,131.022-55.426c28.644-24.568,53.382-45.787,104.98-45.787c51.599,0,76.337,21.219,104.98,45.787 c8.385,7.191,21.011,6.225,28.202-2.159C514.371,370.03,513.405,357.404,505.02,350.213z"
      />
    </svg>
    <p>
      5.3
       mph
    </p>
    <p>
      SSE
    </p>
  </div>
  <p
    className="textBlock"
  >
    "Sunny with a chance of meatballs."
  </p>
</div>
`;

exports[`Single Day Summary Component Should render without errors 1`] = `
<div
  className="SingleDaySummary"
>
  <div
    className="selectionTab"
  />
  <div
    className="weekday"
  >
    <p>
      Monday
    </p>
  </div>
  <div
    className="icon"
  >
    <svg
      height="581pt"
      viewBox="0 0 512 512"
      width="581pt"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M50.087,239.304H16.696C7.475,239.304,0,246.78,0,256s7.475,16.696,16.696,16.696h33.391 c9.22,0,16.696-7.475,16.696-16.696S59.307,239.304,50.087,239.304z"
      />
      <path
        d="M495.304,239.304h-33.391c-9.22,0-16.696,7.475-16.696,16.696s7.475,16.696,16.696,16.696h33.391 c9.22,0,16.696-7.475,16.696-16.696S504.525,239.304,495.304,239.304z"
      />
      <path
        d="M256,445.217c-9.22,0-16.696,7.475-16.696,16.696v33.391c0,9.22,7.475,16.696,16.696,16.696s16.696-7.475,16.696-16.696 v-33.391C272.696,452.693,265.22,445.217,256,445.217z"
      />
      <path
        d="M256,0c-9.22,0-16.696,7.475-16.696,16.696v33.391c0,9.22,7.475,16.696,16.696,16.696s16.696-7.475,16.696-16.696V16.696 C272.696,7.475,265.22,0,256,0z"
      />
      <path
        d="M390.111,454.895l-16.696-28.917c-4.61-7.985-14.821-10.722-22.806-6.111c-7.985,4.61-10.721,14.821-6.111,22.806 l16.696,28.917c4.612,7.989,14.826,10.719,22.806,6.111C391.985,473.091,394.721,462.88,390.111,454.895z"
      />
      <path
        d="M167.502,69.326l-16.696-28.917c-4.61-7.985-14.821-10.722-22.806-6.111c-7.985,4.61-10.721,14.821-6.111,22.806 l16.696,28.917c4.612,7.989,14.826,10.72,22.806,6.111C169.376,87.522,172.112,77.311,167.502,69.326z"
      />
      <path
        d="M471.591,361.194l-28.917-16.696c-7.986-4.612-18.196-1.875-22.806,6.111c-4.61,7.985-1.874,18.196,6.111,22.806 l28.917,16.696c7.981,4.609,18.194,1.878,22.806-6.111C482.312,376.015,479.576,365.804,471.591,361.194z"
      />
      <path
        d="M86.022,138.585l-28.917-16.696c-7.986-4.612-18.197-1.874-22.806,6.111c-4.61,7.985-1.874,18.196,6.111,22.806 l28.917,16.696c7.981,4.609,18.194,1.878,22.806-6.111C96.742,153.406,94.007,143.195,86.022,138.585z"
      />
      <path
        d="M477.702,128c-4.61-7.985-14.82-10.722-22.806-6.111l-28.917,16.696c-7.985,4.61-10.721,14.821-6.111,22.806 c4.612,7.989,14.826,10.719,22.806,6.111l28.917-16.696C479.576,146.196,482.312,135.985,477.702,128z"
      />
      <path
        d="M92.132,350.609c-4.609-7.985-14.821-10.721-22.806-6.111l-28.917,16.696c-7.985,4.61-10.721,14.821-6.111,22.806 c4.612,7.989,14.826,10.719,22.806,6.111l28.917-16.696C94.007,368.805,96.742,358.594,92.132,350.609z"
      />
      <path
        d="M384,34.298c-7.986-4.612-18.196-1.874-22.806,6.111l-16.696,28.917c-4.61,7.985-1.874,18.196,6.111,22.806 c7.981,4.609,18.194,1.879,22.806-6.111l16.696-28.917C394.721,49.12,391.985,38.909,384,34.298z"
      />
      <path
        d="M161.391,419.868c-7.985-4.612-18.197-1.875-22.806,6.111l-16.696,28.917c-4.61,7.985-1.874,18.196,6.111,22.806 c7.981,4.609,18.194,1.878,22.806-6.111l16.696-28.917C172.112,434.689,169.376,424.478,161.391,419.868z"
      />
      <path
        d="M256,100.174c-85.922,0-155.826,69.904-155.826,155.826S170.078,411.826,256,411.826S411.826,341.922,411.826,256 S341.923,100.174,256,100.174z M256,378.435c-67.51,0-122.435-54.924-122.435-122.435S188.49,133.565,256,133.565 S378.435,188.49,378.435,256S323.511,378.435,256,378.435z"
      />
      <path
        d="M256,166.957c-49.099,0-89.043,39.945-89.043,89.043c0,9.22,7.475,16.696,16.696,16.696 c9.22,0,16.696-7.475,16.696-16.696c0-30.687,24.966-55.652,55.652-55.652c9.22,0,16.696-7.475,16.696-16.696 C272.696,174.432,265.22,166.957,256,166.957z"
      />
    </svg>
  </div>
  <div
    className="AverageTemp"
  >
    -67
  </div>
</div>
`;
